src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -978,34 +989,65 @@ public async Task UdpReceiveGetsCanceledByDispose()
                     disposedException = true;
                 }
 
-                if (UsesApm)
-                {
-                    Assert.Null(localSocketError);
-                    Assert.True(disposedException);
-                }
-                else if (UsesSync)
+                try
                 {
-                    Assert.Equal(SocketError.Interrupted, localSocketError);
+                    if (UsesApm)
+                    {
+                        Assert.Null(localSocketError);
+                        Assert.True(disposedException);
+                    }
+                    else if (UsesSync)
+                    {
+                        Assert.Equal(SocketError.Interrupted, localSocketError);
+                    }
+                    else
+                    {
+                        Assert.Equal(SocketError.OperationAborted, localSocketError);
+                    }
+                    break;
                 }
-                else
+                catch
                 {
-                    Assert.Equal(SocketError.OperationAborted, localSocketError);
+                    if (retries-- > 0)
+                    {
+                        continue;
+                    }
+
+                    throw;
                 }
-            }, maxAttempts: 10);
+            }
         }
 
-        [Theory]
-        [InlineData(true)]
-        [InlineData(false)]
-        public async Task TcpReceiveSendGetsCanceledByDispose(bool receiveOrSend)
+        public static readonly TheoryData<bool, bool, bool> TcpReceiveSendGetsCanceledByDispose_Data = new TheoryData<bool, bool, bool>
+        {
+            { true, false, false },
+            { true, false, true },
+            { true, true, false },
+            { false, false, false },
+            { false, false, true },
+            { false, true, false },
+        };
+
+        [Theory(Timeout = 40000)]
+        [MemberData(nameof(TcpReceiveSendGetsCanceledByDispose_Data))]
+        public async Task TcpReceiveSendGetsCanceledByDispose(bool receiveOrSend, bool ipv6Server, bool dualModeClient)
         {
+            if (UsesSync && PlatformDetection.IsRedHatFamily7 &&
+                receiveOrSend && (ipv6Server || dualModeClient))
+            {
+                // The IPV6 synchronous Receive case times out on RedHat7 systems
+                // TODO: open a new issue for this case, if the PR gets accepted.
+                return;
```

Comment:
I tried to use `SkipTestException` here, but it got reported as a failure on CI. Probably doesn't work in async context.

Reply:
We'll be able to remove this when the RHEL7 fix is available, and CI machines are updated for it.

Reply:
I'm not sure if it's exactly the same problem as we had with `Connect`.  The native repro does not hang. See:
 https://github.com/dotnet/runtime/pull/43409#issuecomment-709980516.